homeassistant/components/sensor/__init__.py

```diff
@@ -264,10 +290,17 @@ class SensorEntity(Entity):
     )
     _last_reset_reported = False
     _temperature_conversion_reported = False
+    _sensor_option_unit_of_measurement: str | None = None
 
     # Temporary private attribute to track if deprecation has been logged.
     __datetime_as_string_deprecation_logged = False
 
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        if not self.registry_entry:
+            return
+        self.async_registry_entry_updated()
```

Comment:
Many integrations don't bother calling `super().async_added_to_hass()`, meaning there's a big risk this won't be called.